io/eolearn/io/processing_api.py

```diff
@@ -0,0 +1,205 @@
+        ''' Make a WebFeatureService request to get dates and clean them according to self.time_difference
+        '''
+        wfs = WebFeatureService(
+            bbox=bbox, time_interval=time_interval, data_source=self.data_source, maxcc=self.maxcc
+        )
+
+        dates = wfs.get_dates()
+
+        if len(dates) == 0:
+            raise ValueError("No available images for requested time range: {}".format(time_interval))
+
+        dates = sorted(dates)
+        dates = [dates[0]] + [d2 for d1, d2 in zip(dates[:-1], dates[1:]) if d2 - d1 > self.time_difference]
+        return dates
+
+    @staticmethod
+    def size_from_resolution(bbox, resolution):
+        ''' Calculate size_x and size_y based on provided bbox and resolution
+        '''
+        bbox = list(bbox)
+        size_x = int((bbox[2] - bbox[0]) / resolution)
+        size_y = int((bbox[3] - bbox[1]) / resolution)
+        return size_x, size_y
+
+    def execute(self, eopatch=None, bbox=None, time_interval=None):
+        ''' Make a WFS request to get valid dates, download an image for each valid date and store it in an EOPatch
+
+        :param eopatch: input EOPatch
+        :type eopatch: EOPatch
+        '''
+
+        if self.size is not None:
+            size_x, size_y = self.size
+        elif self.resolution is not None:
+            size_x, size_y = self.size_from_resolution(bbox, self.resolution)
+
+        responses = [shr.response('default', 'image/tiff'), shr.response('userdata', 'application/json')]
+        request = shr.body(
+            request_bounds=shr.bounds(crs=bbox.crs.opengis_string, bbox=list(bbox)),
+            request_data=[shr.data(data_type=self.data_source.api_identifier())],
+            request_output=shr.output(size_x=size_x, size_y=size_y, responses=responses),
+            evalscript=self.generate_evalscript()
+        )
+
+        request_args = dict(
+            url=SHConfig().get_sh_processing_api_url(),
+            headers={"accept": "application/tar", 'content-type': 'application/json'},
+            data_folder=self.cache_folder,
+            hash_save=bool(self.cache_folder),
+            request_type='POST',
+            data_type=MimeType.TAR
+        )
+
+        dates = self.get_dates(bbox, time_interval)
+        requests = (self.request_from_date(request, date, self.maxcc, self.time_difference) for date in dates)
+        requests = [DownloadRequest(post_values=payload, **request_args) for payload in requests]
+
+        LOGGER.debug('Downloading %d requests of type %s', len(requests), str(self.data_source))
+        LOGGER.debug('Downloading bands: [%s]', ', '.join(self.all_bands))
+        client = SentinelHubDownloadClient()
+        images = client.download(requests)
+        LOGGER.debug('Downloads complete')
+
+        images = ((img['default.tif'], img['userdata.json']) for img in images)
+        images = [(img, meta.get('norm_factor', 0) if meta else 0) for img, meta in images]
```

Comment:
I must re-check this, there were some exceptions when we got empty images (all zeros) and in this cases we got None instead of dict as meta, but that might have been in the old way of decoding tars.

Reply:
Seems ok, removed.